command-tunneler/command_tunneler.go

```diff
@@ -1,9 +1,5 @@
 package command_tunneler
 
-import "github.com/odedpriva/cli-transparent-tunnel/config"
-
 type TunnelerCommand interface {
-	GetTunnelConfiguration() (*config.TunnelConfiguration, error)
-	GetCommandWithTunnel(tunnelAddress string, originalServer string) (string, []string)
-	GetPlainCommand() (string, []string)
+	GetCommandWithTunnel(tunnelAddress string, originalServer string) (string, []string, error)
```

Comment:
In order to add a specific command (like redis_command), the developer need to:

- implement struct for his specific command - but who will create it and would know what argument to pass?
- Implement GetCommandWithTunnel which will as I understand always need to perform SplitHostPort and SplitHostPort.

Instead I think TunnelerCommand should be responsible only for convertor of command (general data structure not customizable) to the specific command and thus user need to only implement the convertor part not the data struct of the entire command.
Also, the user will just need to "register" his specific convertor and it will be used when this type will be used.